File: backend/src/main/java/com/project2/global/exception/GlobalExceptionHandler.java
package com.project2.global.exception;

import java.util.stream.Collectors;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.MethodArgumentNotValidException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.multipart.MaxUploadSizeExceededException;

import com.project2.global.dto.RsData;

@RestControllerAdvice
public class GlobalExceptionHandler {

	@ExceptionHandler(MethodArgumentNotValidException.class)
	public ResponseEntity<RsData<Void>> handleMethodArgumentNotValidException(MethodArgumentNotValidException e) {

		String message = e.getBindingResult().getFieldErrors()
			.stream()
			.map(fe -> fe.getField() + " : " + fe.getCode() + " : " + fe.getDefaultMessage())
			.sorted()
			.collect(Collectors.joining("\n"));

		return ResponseEntity
			.status(HttpStatus.BAD_REQUEST)
			.body(
				new RsData<>(
					String.valueOf(HttpStatus.BAD_REQUEST.value()),
					message
				)
			);
	}

	@ResponseStatus
	@ExceptionHandler(ServiceException.class)
	public ResponseEntity<RsData<Void>> ServiceExceptionHandle(ServiceException ex) {

		return ResponseEntity
			.status(ex.getStatusCode())
			.body(
				new RsData<>(
					ex.getCode(),
					ex.getMsg()
				)
			);
	}

	@ExceptionHandler(MaxUploadSizeExceededException.class)
	@ResponseStatus(HttpStatus.PAYLOAD_TOO_LARGE)
	public ResponseEntity<RsData<Void>> handleMaxSizeException(MaxUploadSizeExceededException ex) {

		return ResponseEntity
			.status(ex.getStatusCode())
			.body(
				new RsData<>(String.valueOf(HttpStatus.PAYLOAD_TOO_LARGE.value()),
					"파일 크기가 너무 큽니다. 각 파일은 최대 10MB, 총 50MB까지 업로드 가능합니다.")
			);
	}

}

File: backend/src/main/kotlin/com/project3/global/exception/GlobalExceptionHandler.kt
package com.project3.global.exception

import com.project3.global.dto.RsData
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.MethodArgumentNotValidException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice
import org.springframework.web.multipart.MaxUploadSizeExceededException

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(MethodArgumentNotValidException::class)
    fun handleMethodArgumentNotValidException(e: MethodArgumentNotValidException): ResponseEntity<RsData<Void>> {
        val message = e.bindingResult.fieldErrors
                .joinToString("\n") {
                    "${it.field} : ${it.code} : ${it.defaultMessage}"
                }

        return ResponseEntity
                .status(HttpStatus.BAD_REQUEST)
                .body(
                        RsData(
                                HttpStatus.BAD_REQUEST.value().toString(),
                                message
                        )
                )
    }

    @ExceptionHandler(ServiceException::class)
    @ResponseStatus
    fun handleServiceException(ex: ServiceException): ResponseEntity<RsData<Void>> {
        return ResponseEntity
                .status(ex.statusCode)
                .body(
                        RsData(
                                ex.code,
                                ex.msg
                        )
                )
    }

    @ExceptionHandler(MaxUploadSizeExceededException::class)
    @ResponseStatus(HttpStatus.PAYLOAD_TOO_LARGE)
    fun handleMaxSizeException(ex: MaxUploadSizeExceededException): ResponseEntity<RsData<Void>> {
        return ResponseEntity
                .status(HttpStatus.PAYLOAD_TOO_LARGE)
                .body(
                        RsData(
                                HttpStatus.PAYLOAD_TOO_LARGE.value().toString(),
                                "파일 크기가 너무 큽니다. 각 파일은 최대 10MB, 총 50MB까지 업로드 가능합니다."
                        )
                )
    }
}